File: 303.Range Sum Query - Immutable/303.Range Sum Query - Immutable.java
public NumArray(int[] nums) {
　　　　  //nums[0]不变 num[i]=num[i]+前一位
        for(int i = 1; i < nums.length; i++){
            nums[i] =nums[i]+nums[i - 1];
        }

        this.nums = nums;
    }

    public int sumRange(int i, int j) {
        if(i == 0){
            return nums[j];
        }

        return nums[j] - nums[i - 1];
    }

File: 303.Range Sum Query - Immutable/303.Range Sum Query - Immutable.cpp
class NumArray {
public:
    NumArray(vector<int> nums) {
        dp.resize(nums.size()+1,0);
        for(int i=1;i<=nums.size();++i){
            dp[i] = dp[i-1]+nums[i-1];
        }
    }

    int sumRange(int i, int j) {
        return dp[j+1]-dp[i];
    }
    private: 
    vector<int> dp;
};
